### ISSUE TITLE
Transitional dependencies issue

### ISSUE THREAD

**Issue by pierreis** [UnitedIncome/serverless-python-requirements#68]
I tried to explicitely declare as `nodeploy` the dependencies provided on the Amazon platform.

```
noDeploy:
  - boto3
  - botocore
  - docutils
  - jmespath
  - python-dateutil
  - s3transfer
  - six
```

Yet, this does not seem to work with transitional dependencies.
For example, my project requires the Amazon Encryption SDK, which itself relies on boto3.
The dependencies in questions are packaged and significantly increase the deployment pachage size, despite the `nodeploy`.

Is it a command to avoid deploying transitional dependancies?

**Comment by pierreis** [comment]
Actually, I was using the NPM version, which does not seem to include package filtering.
However, installing the github version, I run in an additional issue:

```
  Error --------------------------------------------------
 
  ENOENT: no such file or directory, unlink 'boto3'
 
     For debugging logs, run again after setting the "SLS_DEBUG=*" environment variable.
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Forums:        forum.serverless.com
     Chat:          gitter.im/serverless/serverless
 
  Your Environment Information -----------------------------
     OS:                     darwin
     Node Version:           8.4.0
     Serverless Version:     1.18.0
```

The noDeploy configuration is the same as above.
Quickly looking at the code, it seems that `unlinkRequirements` fails when trying to unlink packages that were not linked by `linkRequirements`, because they are explicitly ignored.

**Comment by dschep** [comment]
Good catch. It is on npm tagged as `beta` fyi 😉

**Comment by dschep** [comment]
Just released `3.0.0-beta.2` which should fix this
